packages/commons-ui-core/src/RichTypography/RichTypography.js

```diff
@@ -1,39 +1,61 @@
 import { styled } from "@mui/material/styles";
 import Typography from "@mui/material/Typography";
 import PropTypes from "prop-types";
-import * as React from "react";
+import React, { useEffect, useImperativeHandle, useRef } from "react";
 
-const RichTypographyRoot = styled(Typography)(({ theme }) => ({
-  "& a": {
-    color: theme.palette.main,
-  },
-}));
+const RichTypographyRoot = styled(Typography, {
+  shouldForwardProp: (prop) => prop !== "LinkProps",
+})(({ LinkProps, theme }) => {
+  const color = LinkProps?.color || "primary.main";
+  return {
+    "& a, & a:visited": {
+      color: color.split(".").reduce((acc, curr) => acc[curr], theme.palette),
+    },
+  };
+});
 
 const RichTypography = React.forwardRef(function RichTypography(
-  { children, component, ...props },
+  { LinkProps, children: childrenProp, component, ...props },
   ref
 ) {
-  if (!children) {
-    return null;
+  const typographyRef = useRef();
+  useImperativeHandle(ref, () => typographyRef.current);
+  useEffect(() => {
+    if (LinkProps?.onClick) {
+      const { current: el } = typographyRef;
+      if (el) {
+        const anchors = el.getElementsByTagName("a");
+        for (let i = 0; i < anchors.length; i += 1) {
+          anchors[i].onclick = LinkProps.onClick;
+        }
+      }
+    }
+  }, [LinkProps?.onClick]);
+  let children;
+  let dangerouslySetInnerHTML;
+  if (typeof childrenProp === "string") {
+    dangerouslySetInnerHTML = {
```

Comment:
With this, we assume that we trust anyone adding content to the CMS right? Tested this by adding something like 
`<img src=??? onerror="alert('XSS')" />` which gets executed in the F.E

Reply:
Yes and no.

Since it's behind GitHub login, we can say we trust the people adding content. But really, we shouldn't.

But, since you're the one who chose `markedjs` over `Remark`, you should have added `DOMPurify` per their [docs](https://marked.js.org/#usage).

Reply:
yeah yeah. Should it be added to the current PR? 😂

Reply:
I take it we'll be getting a PR adding DOMPurify from you @esirK ?